### PR TITLE
feat: Minor improvements to code style

### PR DESCRIPTION
**Pull request by ijsbol**
Hiya, I made some improvements to the typing along with updating the module to follow PEP best practices such as using `__all__` in the `__init__` file & not shadowing variable types on `SnowflakeGenerator.__init__`. Additionally, I've updated both classes to use `__slots__` which improves overal performance.

### CODE REVIEW

File: src/snowflake/snowflake.py [hunk 1]
```diff
@@ -4,18 +4,26 @@
...
-MAX_TS = 0b11111111111111111111111111111111111111111
-MAX_INSTANCE = 0b1111111111
-MAX_SEQ = 0b111111111111
+MAX_TS: Final[int] = 0b11111111111111111111111111111111111111111
+MAX_INSTANCE: Final[int] = 0b1111111111
+MAX_SEQ: Final[int] = 0b111111111111
 
 
-@dataclass(frozen=True)
+@dataclass(frozen=True, slots=True)
```
Comment (vd2org) [inline src/snowflake/snowflake.py]: It's not compatible with 3.8 and 3.9.

https://docs.python.org/3/library/dataclasses.html#dataclasses.dataclass

File: src/snowflake/snowflake.py [hunk 2]
```diff
@@ -56,7 +64,7 @@ def seconds(self) -> float:
     def datetime(self) -> datetime:
         return datetime.utcfromtimestamp(self.seconds)
 
-    def datetime_tz(self, tz: Optional[tzinfo] = None) -> datetime:
+    def datetime_tz(self, tz: Optional[tzinfo] = None) -> 'datetime':
```
Comment (vd2org) [inline src/snowflake/snowflake.py:67]: Why?

**Review by vd2org — CHANGES_REQUESTED** [review]
Thank you for your contribution. Reviewed.

File: src/snowflake/snowflake.py (hunk 1 above)
Comment (ijsbol) [reply src/snowflake/snowflake.py]: Intresting, good catch.

File: src/snowflake/snowflake.py (hunk 2 above)
Comment (ijsbol) [reply src/snowflake/snowflake.py:67]: Raises type errors in newer Python versions.

**Comment by ijsbol** [comment]
Updated for <3.10 support.

File: src/snowflake/snowflake.py (hunk 2 above)
Comment (vd2org) [reply src/snowflake/snowflake.py:67]: Hmmm... specifically?

Comment (vd2org) [reply src/snowflake/snowflake.py:67]: Checked on 3.11 and 3.12. No errors there. Did I miss something?

Comment (ijsbol) [reply src/snowflake/snowflake.py:67]: It was raising unknown type errors when using pyright for 3.12.2

Comment (vd2org) [reply src/snowflake/snowflake.py:67]: Can't reproduce it. Could you share the full example, please?

**Pull request closed without merging**